test/libsolidity/SyntaxTest.h

```diff
@@ -52,33 +50,31 @@ struct SyntaxTestError
 };
 
 
-class SyntaxTest: AnalysisFramework
+class SyntaxTest: AnalysisFramework, public TestCase
 {
 public:
+	static std::unique_ptr<TestCase> create(std::string const& _filename)
+	{ return std::unique_ptr<TestCase>(new SyntaxTest(_filename)); }
 	SyntaxTest(std::string const& _filename);
 
 	bool run(std::ostream& _stream, std::string const& _linePrefix = "", bool const _formatted = false);
```

Comment:
Please always use override for virtual functions, and perhaps even virtual in the derived class.